src/test/java/io/reactivex/schedulers/TestSchedulerTest.java

```diff
@@ -216,4 +217,40 @@ public void run() {
             inner.dispose();
         }
     }
+
+    @Test
+    public final void testConcurrentInsertionAndTimeChange() {
+        final TestScheduler scheduler = new TestScheduler();
+        new Thread() {
+            @Override public void run() {
+                while (true) {
+                    Thread.yield(); // Scheduling service lock is not fair
+                    final Scheduler.Worker inner = scheduler.createWorker();
+                    inner.schedule(new Runnable() {
+                        @Override
+                        public void run()
+                        {
+                            System.out.println(scheduler.now(TimeUnit.MILLISECONDS));
+                        }
+
+                    },1, TimeUnit.SECONDS);
+                }
+            }
+        }.start();
```

Comment:
Whole test looks very flaky, I don't see any kind of threads joining, it may run differently each time!